realclientip_test.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetRealClientIPParser(t *testing.T) {
+	var p realClientIPParser
+	var err error
+
+	p, err = getRealClientIPParser("X-Forwarded-For")
+	assert.Nil(t, err)
+	assert.Equal(t, reflect.TypeOf(p), reflect.TypeOf((*xForwardedForClientIPParser)(nil)))
```

Comment:
Should we be checking the header value for this and all the others in this test? I wonder if this may be better styled as a table driven test?

Reply:
You're probably right, a table of test cases would be cleaner.